source/core/memory.cpp

```diff
@@ -291,36 +292,33 @@ void Memory::deserializeJSON(const Json& json) {
   }
   // TODO::check types!
   // check that sizes match
-  std::size_t byteCount = *byteCountIt;
-  if (_byteCount != byteCount) {
-    std::string expected = std::to_string(_byteCount);
-    std::size_t byteCount = *byteCountIt;
-    std::string recieved = std::to_string(byteCount);
-    throw DeserializationError(
-        "Could not deserialize Memory: ByteCount did not match: expected \"" +
-        expected + "\" to be equal to \"" + recieved + "\"");
+  size_t byteCount = *byteCountIt;
+  if (_byteCount < byteCount) {
+    _byteCount = byteCount;
+    _data = MemoryValue(byteCount * _byteSize);
+    _sizeCallback(_byteCount);
   }
```

Comment:
Reducing the size of the memory can cause problems with gui components, which might not adapt to the new memory size immediately. This is not critical when increasing the size, but can cause crashes when decreasing the size.
Because of that, the size is only changed if the snapshot would not fit into the current memory.

It's also part of the description of this pull request...